src/main/java/org/codehaus/plexus/archiver/UnArchiver.java

```diff
@@ -72,6 +73,25 @@ void extract( String path, File outputDirectory )
      */
     void setOverwrite( boolean b );
 
+    /**
+     * Get component which rewrites the target path of each unpacked file.
+     *
+     * @return {@link FileMapper} to be used for rewriting each target path, or {@code null} if no rewriting shall happen.
+     *
+     * @since 3.7
+     */
+    FileMapper getFileMapper();
+
+    /***
+     * Sets component to be used for rewriting target path of each unpacked file.
+     *
+     * @param fileMapper {@link FileMapper} to be used for rewriting each target path, or {@code null} if no
+     * rewriting shall happen.
+     *
+     * @since 3.7
+     */
+    void setFileMapper( FileMapper fileMapper );
```

Comment:
Both `PlexusIoFileResourceCollection` and `BaseFileSet` accept an array of `FileMapper`, so it would be more flexible and consistent if `UnArchiver` allows the user to set an array as well.

Reply:
What shall be the outcome? Currently I rewrite the path so /a/b/c becomes /x/y/z. If I apply several filemappers, that would multipy the resulting paths. But having multiple copies is a *different feature*.

Reply:
Thanks. I do agree that the only file mapper needed is `RegExMapper` but I really think having the ability to pass multiple simple file mappers would make it easier to use (and it would be consistent with the rest of API as I mentioned).

Reply:
Agreed. I will provide this in the next days.

Reply:
Fixed by https://github.com/mkarg/plexus-archiver/commit/93762b38eff29b9988252f4085e9185092ccec01.